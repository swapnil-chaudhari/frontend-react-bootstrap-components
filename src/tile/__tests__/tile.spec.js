import React from 'react';
import renderShallow from 'render-shallow';
import noop from 'src/utils/noop';
import Tile from '../tile';

describe('<Tile>', () => {
    describe('when it renders without image', () => {
        const props = {
            children: <div />,
            onClick: noop
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Tile { ...props } />).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with image', () => {
        const props = {
            image: 'https://google.com/images/1',
            children: <div />,
            onClick: noop
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Tile { ...props } />).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
