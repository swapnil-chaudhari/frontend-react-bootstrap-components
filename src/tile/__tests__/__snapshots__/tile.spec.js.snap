// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile> when it renders with image has proper markup 1`] = `
<div
  className="Tile"
  onClick={[Function]}
>
  <div
    className="row"
  >
    <div
      className="Tile-image-column col-xs-4"
    >
      <div
        className="Tile-image-column-inner"
      />
    </div>
    <div
      className="Tile-column col-xs-8"
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`<Tile> when it renders without image has proper markup 1`] = `
<div
  className="Tile"
  onClick={[Function]}
>
  <div
    className="row"
  >
    <div
      className="Tile-column col-xs-12"
    >
      <div />
    </div>
  </div>
</div>
`;
